src/components/SingerList.jsx

```diff
@@ -5,20 +5,20 @@ import {Button} from "react-bootstrap";
 
 
 export class SingerList extends React.Component{
-    constructor(){
-        super();
+    constructor(props){
+        super(props);
         this.state={
           data:[],
           count:0,
           next_page:"",
           previous_page:"",
-          current_page:"/singers/"
+          current_page:"?page=1",
```

Comment:
I think this should be done using query params with router

https://reactrouter.com/web/example/query-parameters